### ISSUE TITLE
Fedora 21 mostly working but NullPointerException at Karaf shell

### ISSUE THREAD

**Issue by dfarrell07** [dfarrell07/vagrant-opendaylight#20]
As described [here](https://github.com/dfarrell07/vagrant-opendaylight/issues/16#issuecomment-76978081), I'm seeing an unexpected NullPointerException from the `feature:list` Karaf shell command on Fedora 21 boxes (Puppet-tarball method at least, checking direct RPM and Puppet-RPM now).

```
opendaylight-user@root>feature:list
Error executing command: java.lang.NullPointerException
```

ODL seems to _mostly_ work. It's installed, the service is running and active, it accepts connections.

```
[~/vagrant-opendaylight]$ vagrant ssh fed21_puppet_tarball
Last login: Tue Mar  3 16:05:21 2015 from 10.0.2.2
[vagrant@localhost ~]$ ls /opt/
opendaylight-0.2.2  VBoxGuestAdditions-4.3.20
[vagrant@localhost ~]$ sudo systemctl status opendaylight
● opendaylight.service - OpenDaylight SDN Controller
   Loaded: loaded (/usr/lib/systemd/system/opendaylight.service; enabled)
   Active: active (running) since Tue 2015-03-03 16:16:02 UTC; 2min 13s ago
     Docs: https://wiki.opendaylight.org/view/Main_Page
           http://www.opendaylight.org/
 Main PID: 2790 (java)
   CGroup: /system.slice/opendaylight.service
           └─2790 /usr/bin/java -server -Xms128M -Xmx2048m -XX:+UnlockDiagnosticVMOptio...

Mar 03 16:16:02 localhost.localdomain systemd[1]: Started OpenDaylight SDN Controller.
[vagrant@localhost ~]$ pgrep java
2790
[vagrant@localhost ~]$ cd /vagrant/scripts/
[vagrant@localhost scripts]$ ./connect.sh 
# snip sshpass install
Will repeatedly attempt connecting to Karaf shell until it's ready
Warning: Permanently added '[localhost]:8101' (DSA) to the list of known hosts.
Authenticated with partial success.

    ________                       ________                .__  .__       .__     __       
    \_____  \ ______   ____   ____ \______ \ _____  ___.__.|  | |__| ____ |  |___/  |_     
     /   |   \\____ \_/ __ \ /    \ |    |  \\__  \<   |  ||  | |  |/ ___\|  |  \   __\    
    /    |    \  |_> >  ___/|   |  \|    `   \/ __ \\___  ||  |_|  / /_/  >   Y  \  |      
    \_______  /   __/ \___  >___|  /_______  (____  / ____||____/__\___  /|___|  /__|      
            \/|__|        \/     \/        \/     \/\/            /_____/      \/          


Hit '<tab>' for a list of available commands
and '[cmd] --help' for help on a specific command.
Hit '<ctrl-d>' or type 'system:shutdown' or 'logout' to shutdown OpenDaylight.

opendaylight-user@root>
```

**Comment by dfarrell07** [comment]
I'm able to reproduce this failure in the F21 box provisioned by a direct ODL RPM install.

```
[~/vagrant-opendaylight]$ vagrant ssh f21_rpm
Last login: Thu Dec 18 07:06:12 2014 from 10.0.2.2
[vagrant@localhost ~]$ ls /opt/
opendaylight-0.2.2  VBoxGuestAdditions-4.3.20
[vagrant@localhost ~]$ sudo systemctl status opendaylight
● opendaylight.service - OpenDaylight SDN Controller
   Loaded: loaded (/usr/lib/systemd/system/opendaylight.service; disabled)
   Active: active (running) since Tue 2015-03-03 19:01:29 UTC; 5min ago
     Docs: https://wiki.opendaylight.org/view/Main_Page
           http://www.opendaylight.org/
  Process: 1838 ExecStart=/opt/opendaylight-0.2.2/bin/start (code=exited, status=0/SUCCESS)
 Main PID: 1845 (java)
   CGroup: /system.slice/opendaylight.service
           └─1845 /usr/bin/java -server -Xms128M -Xmx2048m -XX:+UnlockDiagnosticVMOptio...

Mar 03 19:01:29 localhost.localdomain systemd[1]: Started OpenDaylight SDN Controller.
[vagrant@localhost ~]$ cd /vagrant/
[vagrant@localhost vagrant]$ ./scripts/connect.sh 
# snip sshpass install
Will repeatedly attempt connecting to Karaf shell until it's ready
Warning: Permanently added '[localhost]:8101' (DSA) to the list of known hosts.
Authenticated with partial success.

    ________                       ________                .__  .__       .__     __       
    \_____  \ ______   ____   ____ \______ \ _____  ___.__.|  | |__| ____ |  |___/  |_     
     /   |   \\____ \_/ __ \ /    \ |    |  \\__  \<   |  ||  | |  |/ ___\|  |  \   __\    
    /    |    \  |_> >  ___/|   |  \|    `   \/ __ \\___  ||  |_|  / /_/  >   Y  \  |      
    \_______  /   __/ \___  >___|  /_______  (____  / ____||____/__\___  /|___|  /__|      
            \/|__|        \/     \/        \/     \/\/            /_____/      \/          


Hit '<tab>' for a list of available commands
and '[cmd] --help' for help on a specific command.
Hit '<ctrl-d>' or type 'system:shutdown' or 'logout' to shutdown OpenDaylight.

opendaylight-user@root>feature:list
Error executing command: java.lang.NullPointerException
opendaylight-user@root>
```

**Comment by dfarrell07** [comment]
This is caused by an incompatibility with openjdk-1.8 (used by Fedora >=21) and Karaf 3.0.1 (used by ODL 0.2.3). Using a build of ODL from master (so Lithium) would work with Java 8 (as it uses Karaf 3.0.3), but we currently don't have a way to install such a build via an RPM. The other option is to hack an install of Java 7 onto an F21 VM, but that [seems pretty ugly](https://ask.fedoraproject.org/en/question/61878/how-can-i-to-install-openjdk-7-on-fedora-21/).

tl;dr is that there isn't a good way to run the latest release of ODL on Fedora 21 or 22 at the moment.

Note that Fedora 22 was released today, which implies that Fedora 20 will go end of life in 1 month.
